drivers/resolver/ad2s1210/ad2s1210.c

```diff
@@ -0,0 +1,408 @@
+
+	ret = no_os_gpio_get_optional(&d->gpio_a0, &init_param->gpio_a0);
+	if (ret)
+		goto err_a0;
+
+	ret = no_os_gpio_get_optional(&d->gpio_a1, &init_param->gpio_a1);
+	if (ret)
+		goto err_a1;
+
+	ret = no_os_gpio_get_optional(&d->gpio_res0, &init_param->gpio_res0);
+	if (ret)
+		goto err_res0;
+
+	ret = no_os_gpio_get_optional(&d->gpio_res1, &init_param->gpio_res1);
+	if (ret)
+		goto err_res1;
+
+	ret = no_os_gpio_direction_output(d->gpio_a0, NO_OS_GPIO_HIGH);
+	if (ret)
+		goto err_output;
+
+	ret = no_os_gpio_direction_output(d->gpio_a1, NO_OS_GPIO_HIGH);
+	if (ret)
+		goto err_output;
+
+	ret = no_os_gpio_direction_output(d->gpio_res0, NO_OS_GPIO_HIGH);
+	if (ret)
+		goto err_output;
+
+	ret = no_os_gpio_direction_output(d->gpio_res1, NO_OS_GPIO_HIGH);
+	if (ret)
+		goto err_output;
+
+	ret = no_os_gpio_direction_output(d->gpio_sample, NO_OS_GPIO_HIGH);
+	if (ret)
+		goto err_output;
+
+	ret = no_os_spi_init(&d->spi_desc, &init_param->spi_init);
+	if (ret)
+		goto err_output;
+
+	if (d->gpio_a0 && d->gpio_a1)
+		d->have_mode_pins = true;
+
+	if (d->gpio_res0 && d->gpio_res1)
+		d->have_resolution_pins = true;
+
+	ret = ad2s1210_set_resolution(d, d->resolution);
+	if (ret)
+		goto err_spi;
+
+	*dev = d;
+	return 0;
+
+err_spi:
+	no_os_spi_remove(d->spi_desc);
+err_output:
+	no_os_gpio_remove(d->gpio_res1);
+err_res1:
+	no_os_gpio_remove(d->gpio_res0);
+err_res0:
+	no_os_gpio_remove(d->gpio_a1);
+err_a1:
+	no_os_gpio_remove(d->gpio_a0);
+err_a0:
+	no_os_gpio_remove(d->gpio_sample);
+err_sample:
+	no_os_free(d);
+	pr_err("%s initialization failed with status %ld\n", d->name, ret);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Returns the result of a single conversion.
+ *
+ * @param dev - The device structure.
+ * @param chn - Channel to use angle = 0, velocity = 1
+ * @param data - Buffer to store sampled register data
+ *
+ * @return 0 in case of success or negative error code.
+*******************************************************************************/
+int ad2s1210_spi_single_conversion(struct ad2s1210_dev *dev,
+				   enum ad2s1210_channel chn,
+				   uint16_t *data)
+{
+	int32_t ret;
+	uint8_t addr = AD2S1210_REG_POSITION;
+	enum ad2s1210_mode mode = MODE_POS;
+
+	ret = no_os_gpio_set_value(dev->gpio_sample, NO_OS_GPIO_LOW);
+	if (ret)
+		return ret;
+
+	ret = no_os_gpio_set_value(dev->gpio_sample, NO_OS_GPIO_HIGH);
+	if (ret)
+		return ret;
+
+	if (dev->have_mode_pins) {
+		if (chn == AD2S1210_VEL)
+			mode = MODE_VEL;
+
+		ret = ad2s1210_set_mode_pins(dev, mode);
+		if (ret)
+			return ret;
+
+		ret = no_os_spi_write_and_read(dev->spi_desc, (uint8_t *)data, 2);
+		if (ret)
+			return ret;
+
+	} else {
+		if (chn == AD2S1210_VEL)
+			addr = AD2S1210_REG_VELOCITY;
+
+		ret = ad2s1210_reg_read(dev, addr, (uint8_t *)data);
+		if (ret)
+			return ret;
+
+		ret = ad2s1210_reg_read(dev, addr + 1, (uint8_t *)data + 1);
+		if (ret)
+			return ret;
+	}
+
+	return 0;
```

Comment:
This selection can be rewritten without the else... it will save some lines of code.
In the if statement simply return no_os_spi_write_and_read, then continue with if and return directly ad2s1210_reg_read